Sources/Rendering/Core/Picker/index.js

```diff
@@ -89,9 +89,171 @@ function vtkPicker(publicAPI, model) {
     return Math.sqrt(tolerance);
   }
 
+  /**
+   * Perform picking on the given renderer, given a ray defined in world coordinates.
+   * @param {*} renderer
+   * @param {*} tolerance
+   * @param {*} p1World
+   * @param {*} p2World
+   * @returns true if we picked something else false
+   */
+  function pick3DInternal(renderer, tolerance, p1World, p2World) {
+    const p1Mapper = new Float64Array(4);
+    const p2Mapper = new Float64Array(4);
+    const t = [];
+
+    const ray = [];
+    let pickable;
+    const hitPosition = [];
+
+    if (!renderer) {
+      vtkErrorMacro('You must specify a renderer.');
+      return;
+    }
```

Comment:
`renderer` already checked in the calling methods. You must decide whose responsibility is it to test for renderer validity, but do not do it more than needed